apps/api/src/app/workflows-v2/usecases/build-variable-schema/build-available-variable-schema.usecase.ts

```diff
@@ -39,14 +44,35 @@ export class BuildAvailableVariableSchemaUsecase {
           additionalProperties: false,
         },
         steps: buildPreviousStepsSchema(previousSteps, workflow.payloadSchema),
-        payload: parsePayloadSchema(workflow.payloadSchema, { safe: true }) || {
-          type: 'object',
-          description: 'Payload for the current step',
-        },
+        payload: await this.resolvePayloadSchema(workflow, command),
       },
       additionalProperties: false,
     } as const satisfies JSONSchemaDto;
   }
+
+  @Instrument()
+  private async resolvePayloadSchema(
+    workflow: NotificationTemplateEntity,
+    command: BuildAvailableVariableSchemaCommand
+  ): Promise<JSONSchemaDto> {
+    if (workflow.payloadSchema) {
+      return (
+        parsePayloadSchema(workflow.payloadSchema, { safe: true }) || {
+          type: 'object',
```

Comment:
This parsePayloadSchema invocation is different from the one triggered in apps/api/src/app/workflows-v2/usecases/build-test-data/build-workflow-test-data.usecase.ts. If not mistaken, they should be aligned.